app/controllers/sessions_controller.rb

```diff
@@ -1,4 +1,6 @@
 class SessionsController < ApplicationController
+  before_action :valid_cart?, :only => [:remove_from_cart, :clear_cart, :show_cart]
```

Comment:
Shouldn't these actions be in their own `cart_controller`.
`sessions_controller` seems like it ought to only hold session-related actions, e.g., logging in/out, etc.